provider/pkg/gen/schema.go

```diff
@@ -490,6 +490,31 @@ func (g *packageGenerator) genTypeSpec(prop *discovery.JsonSchema, isOutput bool
 		}, nil
 	case prop.Type == "any":
 		return &schema.TypeSpec{Ref: "pulumi.json#/Any"}, nil
+	case prop.Type == "object" && len(prop.Properties) > 0:
+		tok := fmt.Sprintf(`%s:%s:%s%s`, g.pkg.Name, g.mod, typeName, strings.Title(propName))
+		if isOutput {
+			tok += "Response"
+		}
+		if g.visitedTypes.Has(tok) {
```

Comment:
I think we want to check if `g.rest.schemas[fmt.Sprintf("%s%s", typeName, strings.Title(propName))]`has something instead? `visitedTypes` may not be sufficiently populated to detect a collision yet?

Reply:
Yes, great point! Fixing...